harness/setup.py

```diff
@@ -14,7 +14,7 @@
     python_requires=">=3.6",
     package_data={"determined": ["py.typed"]},
     install_requires=[
-        "cloudpickle==0.5.3",
+        "cloudpickle",
```

Comment:
is cloudpickle forward-compatible?  like can you safely load a checkpoint saved with old cloudpickle using new cloudpickle?

Reply:
That's a very good question. As a sanity check, I did create a TF / Keras checkpoint in the old image and ensured I could continue training in the new one. I'll see if I can get a definitive answer though.

For context, the pressure to move off of 0.5.3 is that it will not work with Python 3.8.

Reply:
ok, but I'm pretty sure only pytorch used cloudpickle anyway, so I'm a bit more interested in that case.  Like basically, can I continue training from my 0.14.3 checkpoint in a 0.14.4 cluster?  That would be a pretty big breakage if it was not true.

Reply:
I've tested a bunch of PyTorch models as well - CIFAR and the NLP models. No problems. You can't go backwards, of course, because of the PyTorch archive version issues that we've sometimes hit in tests when upgrading the taskenv but not the test driver.

There unfortunately doesn't seem to be much of a cloudpickle community - I could find a mailing list or anything, so I filed an issue a few days ago to clarify the *intended* support but haven't heard anything.